datapipelines/src/main/kotlin/io/customer/datapipelines/migration/TrackingMigrationPlugin.kt

```diff
@@ -0,0 +1,34 @@
+package io.customer.datapipelines.migration
+
+import com.segment.analytics.kotlin.core.Analytics
+import com.segment.analytics.kotlin.core.BaseEvent
+import com.segment.analytics.kotlin.core.platform.Plugin
+import com.segment.analytics.kotlin.core.utilities.putInContextUnderKey
+
+/**
+ * Plugin responsible for adding device context extras to the events.
+ * This is because the device context extras cannot be added directly to the events
+ * and need to be added at plugin level.
+ */
+class TrackingMigrationPlugin : Plugin {
+    override val type: Plugin.Type = Plugin.Type.Before
+    override lateinit var analytics: Analytics
+
+    private val deviceContextExtras: MutableMap<BaseEvent, Map<String, String>> = mutableMapOf()
```

Comment:
This might not be the best idea but we don't really have any identifier for an event that doesn't change. Happy to try out other options if there are more ideas.

Reply:
The only other idea i could think of is to probably add them in properties with a unique name and then in the plugin move them from properties to context, that would probably save from keeping the map.

Reply:
That's one good idea to think about. Not sure about the challenges it comes with, but will surely try this out later to reduce cost of keeping this map.